### DISCUSSION TITLE
Support of std::valarray

### DISCUSSION THREAD

**Discussion by kpolitis-msim** [BlueBrain/HighFive#824] [Questions] [answered]
Hello,

I was wondering what the required modifications are to higfive in order to support std::valarray for read/write operations. 
I'm guessing the related work begins with the details::inspector. 

Thanks :-)

**Comment by 1uc** [comment, marked as answer]
That does seem like a useful one to support. Thanks for making us aware.

That's correct, you'd start in `details::inspector`. I'd suggest copy-pasting and modifying this specialization:
https://github.com/BlueBrain/HighFive/blob/d5c756b68748f6644e711283bb744cff659cf022/include/highfive/bits/H5Inspector_misc.hpp#L482
Maybe look at the version for `std::vector` to see how the resizing comes into it. With any luck the inspector is the only trait you need to modify; once that's correct, it is supposed to work.

For setting up tests probably look here:
https://github.com/BlueBrain/HighFive/blob/master/tests/unit/test_all_types.cpp
and add a block for `std::valarray<Scalar>` and then other things like `std::valarray<std::vector<T>>` (if that's permitted).

Or start by adding a simple `TEST_CASE` which checks on things; then expand. Unfortunately, these have now grown enough to require some restructuring.
